mobile/native/src/db/models.rs

```diff
@@ -133,41 +133,70 @@ impl Order {
         order_id: String,
         status: (OrderState, Option<f32>, Option<FailureReason>),
         conn: &mut SqliteConnection,
-    ) -> Result<()> {
-        conn.transaction::<(), _, _>(|conn| {
-            let effected_rows = diesel::update(orders::table)
+    ) -> Result<Order> {
+        let res = conn.exclusive_transaction::<Order, _, _>(|conn| {
+            let order: Order = orders::table
                 .filter(schema::orders::id.eq(order_id.clone()))
-                .set(schema::orders::state.eq(status.0))
-                .execute(conn)?;
+                .first(conn)?;
 
-            if effected_rows == 0 {
-                bail!("Could not update order")
+            tracing::info!("Got here");
```

Comment:
🙃 me too 😝

Reply:
that's why `dbg!` messages are best for debugging  - and clippy can help with removal :p